orchestrator/careplancontributor/service.go

```diff
@@ -438,31 +442,22 @@ func (s Service) authorizeScpMember(request *http.Request) (*ScpValidationResult
 		return nil, err
 	}
 
-	// Use extract CarePlan ID to be used for our query that will get the CarePlan and CareTeam in a bundle
-	var bundle fhir.Bundle
-	err = cpsFHIRClient.Search("CarePlan", url.Values{"_id": {carePlanId}, "_include": {"CarePlan:care-team"}}, &bundle)
+	var carePlan fhir.CarePlan
+
+	err = cpsFHIRClient.ReadWithContext(request.Context(), "CarePlan/"+carePlanId, &carePlan)
 	if err != nil {
-		return nil, err
-	}
+		var outcomeError fhirclient.OperationOutcomeError
 
-	if len(bundle.Entry) == 0 {
-		return nil, coolfhir.NewErrorWithCode("CarePlan not found", http.StatusNotFound)
-	}
+		if errors.As(err, &outcomeError); err != nil {
+			return nil, coolfhir.NewErrorWithCode(outcomeError.Error(), outcomeError.HttpStatusCode)
+		}
 
-	var careTeams []fhir.CareTeam
-	err = coolfhir.ResourcesInBundle(&bundle, coolfhir.EntryIsOfType("CareTeam"), &careTeams)
-	if err != nil {
 		return nil, err
 	}
-	if len(careTeams) == 0 {
-		return nil, coolfhir.NewErrorWithCode("CareTeam not found in bundle", http.StatusNotFound)
-	}
-
-	var carePlan fhir.CarePlan
-	err = coolfhir.ResourceInBundle(&bundle, coolfhir.EntryIsOfType("CarePlan"), &carePlan)
 
+	careTeam, err := coolfhir.CareTeamFromCarePlan(&carePlan)
 	if err != nil {
-		return nil, err
+		return nil, coolfhir.NewErrorWithCode("unable to resolve CareTeam", http.StatusNotFound)
```

Comment:
Hmm, maybe InternalServerError is better? Since there shouldn't be CarePlans without CareTeam. 4xx indicates the client can do something to fix it